### PR TITLE
fix(Grid): Add border to rendered height calc.

### LINKED ISSUE #16640: Resizing calculations of `IgxGrid` with 100% width/height cause infinite grow if the grid has css border applied
## Description  
Adding border to igxgrid causes infinite growth.

 * igniteui-angular version: all supported
 * browser: n/a

## Steps to reproduce  

1. Open and run the attached sample

## Result  
The grid continues to expand downwards

## Expected result  
Not to do that

## Attachments  

[vvjckf49--run.zip](https://github.com/user-attachments/files/24105580/vvjckf49--run.zip)

### PR DESCRIPTION
**Pull request by MayaKirova**
Closes https://github.com/IgniteUI/igniteui-angular/issues/16640
When a grid uses `height: 100%` inside a fixed-height container and has a CSS border applied, the body height calculation was not accounting for the border widths, causing the rendered height to exceed the container and triggering a continuous reflow/growth loop.

## Changes

- **Unit test** (`grid.component.spec.ts`): Verifies that applying a top/bottom border to the grid element correctly reduces `calcHeight` by the combined border width (e.g. 2px top + 2px bottom → `calcHeight` decreases by 4px), confirming the fix prevents the reflow loop.

```typescript
// Apply a 2px border (top and bottom) to the grid's native element
grid.nativeElement.style.borderTop = '2px solid black';
grid.nativeElement.style.borderBottom = '2px solid black';

grid.reflow();
fix.detectChanges();

// Border widths must be subtracted from available body height
expect(grid.calcHeight).toBe(calcHeightNoBorder - 4);
```

### CODE REVIEW

**Review by IMinchev64 — DISMISSED** [review]

**Comment by dkamburov** [comment]
@MayaKirova please add a pr for 21.1.x

**Comment by ChronosSF** [comment]
@MayaKirova , could you please resolve the copilot comments and open a pr to 21.1.x too? (think it's missing for this bugfix)

File: projects/igniteui-angular/grids/grid/src/grid-base.directive.ts [hunk 1]
```diff
@@ -7246,20 +7246,24 @@ export abstract class IgxGridBaseDirective implements GridType,
         if (!this._height) {
             return null;
         }
+        const styles = this.document.defaultView.getComputedStyle(this.nativeElement);
```
Comment (mddragnev) [inline projects/igniteui-angular/grids/grid/src/grid-base.directive.ts]: You can use `gridComputedStyles` because it is already fetched on init instead of fetching it each time on grid recaluclation.

**Review by dkamburov — DISMISSED** [review]

**Comment by dkamburov** [comment]
Missing unit test about the fix, covering the scenario, it was commented here as well - https://github.com/IgniteUI/igniteui-angular/pull/17157#discussion_r3032821822

**Comment by MayaKirova** [comment]
> Missing unit test about the fix, covering the scenario, it was commented here as well - [#17157 (comment)](https://github.com/IgniteUI/igniteui-angular/pull/17157#discussion_r3032821822)

Since this is a rather edge case scenario I didn't feel the need to add a test. But since you insist @copilot add a test for this scenario.

**Review by dkamburov — APPROVED** [review]
